osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+      break;
+    default:
+      LOG(ERROR) << "Attempting to process unsupported crash dump stream: "
+                 << stream;
+      break;
+    }
+  }
+
+  auto dumpBaseAddr = static_cast<unsigned char*>(dumpBase);
+  // Process dump info
+  // First, run functions that store information for later processing
+  unsigned long long dumpFlags = 0;
+  unsigned int tid = 0;
+  unsigned int exCode = 0;
+  unsigned long long exAddr = 0;
+  TEB* teb = nullptr;
+  PEB* peb = nullptr;
+  RTL_USER_PROCESS_PARAMETERS* params = nullptr;
+  logAndStoreDumpType(header, dumpFlags, r);
+  logAndStoreTID(exceptionStream, tid, r);
+  storeTEBAndPEB(threadStream, memoryStream, dumpBaseAddr, tid, teb, peb);
+  storeProcessParams(memoryStream, dumpBaseAddr, peb, params);
+  logAndStoreExceptionCodeAndAddr(exceptionStream, exCode, exAddr, r);
+
+  // Then, process everything else
+  r["crash_path"] = fileName;
+  logDumpTime(header, r);
+  logExceptionMsg(exceptionStream, exCode, exAddr, r);
+  logRegisters(exceptionStream, dumpBaseAddr, r);
+  logPID(miscStream, r);
+  logProcessCreateTime(miscStream, r);
+  logOSVersion(systemStream, r);
+  logPEPathAndVersion(moduleStream, dumpBaseAddr, r);
+  logCrashedModule(moduleStream, dumpBaseAddr, exAddr, r);
+  logBeingDebugged(peb, r);
+  logProcessCmdLine(memoryStream, dumpBaseAddr, params, r);
+  logProcessCurDir(memoryStream, dumpBaseAddr, params, r);
+  logProcessEnvVars(memoryStream, dumpBaseAddr, params, r);
+
+  // Cleanup
+  UnmapViewOfFile(dumpBase);
+  CloseHandle(dumpMapFile);
+  CloseHandle(dumpFile);
+  return;
+}
+
+QueryData genCrashLogs(QueryContext& context) {
+  QueryData results;
+  std::string dumpFolderLocation;
+
+  // Query registry for crash dump folder
+  std::string dumpFolderQuery = "SELECT data FROM registry WHERE key = \"" +
+                                kLocalDumpsRegKey + "\" AND path = \"" +
+                                kDumpFolderRegPath + "\"";
+  SQL dumpFolderResults(dumpFolderQuery);
+
+  if (dumpFolderResults.rows().empty()) {
+    LOG(WARNING)
+        << "No crash dump folder found in registry; using fallback location of "
+        << kFallbackFolder;
+    dumpFolderLocation = kFallbackFolder;
+  } else {
+    RowData dumpFolderRowData = dumpFolderResults.rows()[0].at("data");
+    dumpFolderLocation = dumpFolderRowData;
+  }
+
+  // Fill in any environment variables
+  char expandedDumpFolderLocation[MAX_PATH];
+  ExpandEnvironmentStrings(
+      dumpFolderLocation.c_str(), expandedDumpFolderLocation, MAX_PATH);
+
+  if (!fs::exists(expandedDumpFolderLocation) ||
+      !fs::is_directory(expandedDumpFolderLocation)) {
+    LOG(ERROR) << "Invalid crash dump directory: "
+               << expandedDumpFolderLocation;
+    return results;
+  }
+
+  // Enumerate and process crash dumps
+  fs::directory_iterator iterator(expandedDumpFolderLocation);
+  fs::directory_iterator endIterator;
+  while (iterator != endIterator) {
```

Comment:
Lastly, rather than using `directory_iterators`, prefer something like how the [darwin crashes table](https://github.com/facebook/osquery/blob/master/osquery/tables/system/darwin/crashes.cpp#L161-L162) enumerates it's crash dumps folder. It's a little more succinct and avoids the need for a `while` by prefering a foreach.